src/utils.ts

```diff
@@ -79,3 +80,21 @@ export namespace Option {
     return x !== undefined
   }
 }
+
+export function createLogger<T>(atom: Atom<T>, name?: string): Atom<T> {
```

Comment:
Couple suggestions.

1. Let's move this to the `Atom` namespace, so that the usage looks like this: `Atom.log(state.lens(...))`. This way we will not be polluting the local scope when importing, as `createLogger` can mean a lot of things in different contexts.

2. Change the type of atom parameter to `ReadOnlyAtom<T>` – there is no reason we would need to `set` an atom from the logger, and with this type sig you can also log read only atoms.

3. Let's also make an overload with this type signature: `log<T>(atom: ReadOnlyAtom<T>, log: (prevState: T, newState: T) => void)`, so that we can parametrize the logging (e.g. log not to console but to somewhere else). Then it will also become easier to write a test for this.

What do you think?